### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ibooker**
Update dependencies

-   Update node to v18
-   Update to latest version of TypeScript
-   Update to latest version of prettier & eslint-plugin-prettier, ran prettier
-   Minor updates for other dependencies

For internal tracking -> 2912

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# UNRELEASED
```
Comment (oscarleonnogales) [inline CHANGELOG.md:1]: Nit: Missing a blank line

**Review by oscarleonnogales — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
+
+- Dependency Updates
```
Comment (jplukarski) [inline CHANGELOG.md:3]: Separate by `DevDependencies` and `Dependencies.

**Review by jplukarski — CHANGES_REQUESTED** [review]
Need to also update the node version using in the CI workflow

**Comment by ibooker** [comment]
> Need to also update the node version using in the CI workflow

Had to create a [separate PR](https://github.com/braintree/extended-promise/pull/17) to update the workflow.

**Review by jplukarski — APPROVED** [review]

**Pull request merged**